src/rkr/tracing/Thread.cc

```diff
@@ -1716,6 +1739,20 @@ void Thread::_waitid(Build& build,
                      siginfo_t* infop,
                      int options) noexcept {
   LOGF(trace, "{}: waitid(...)", *this);
-  FAIL << "waitid syscall is not handled yet";
-  resume();
-}
+    
+      finishSyscall([=](Build& build, const IRSource& source, long rc) {
+     resume();
+    
+     // If the syscall failed or returned immediately after WNOHANG, stop processing
+     if (rc <= 0) return;
```

Comment:
Similarly to the above, is this code similar to the handling of some other syscall?

Reply:
It is similar to the wait call as above, I just implemented waitId as how wait4 (above) was implemented.